### PR TITLE
chore: Don't test with node 10

### PR DESCRIPTION
**Pull request by nickmccurdy**
**What**: Don't test with node 10

**Why**: Node 10 reached EOL on July 30th so let's free up some CI resources.

**How**: Remove 10 from node test matrix.

**Checklist**:

- ~[ ]~ Documentation added to the
      [docs site](https://github.com/testing-library/testing-library-docs)
- [x] Tests
- ~[ ]~ Typescript definitions updated
- [x] Ready to be merged

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
I think it's best to wait until ESlint drops support for older versions before we do too

Ref: https://github.com/eslint/eslint/issues/14023

**Comment by nickmccurdy** [comment]
It seems like ESLint has dropped it as of August 5th.

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
 
     strategy:
       matrix:
-        node: [10.12, 10, '12.0', 12, 14, '16.8']
```
Comment (nickmccurdy) [inline .github/workflows/pipeline.yml:54]: Note that Node 16.9.1 fixes the regression in 16.9.0, but we still want to include newer versions of 16 in the future

File: .github/workflows/pipeline.yml [hunk 2]
```diff
@@ -51,7 +51,7 @@ jobs:
...
+        node: [12, 14, 16]
```
Comment (nickmccurdy) [inline .github/workflows/pipeline.yml:54]: Note that Node 16.9.1 fixes the regression in 16.9.0, but we still want to include newer versions of 16 in the future

**Comment by nickmccurdy** [comment]
Does merging this depend on #462?

**Comment by Belco90** [comment]
> Does merging this depend on #462?

That's right. We can point this MR to the alpha branch for the next major, right? @MichaelDeBoey

**Comment by MichaelDeBoey** [comment]
Going to close this one as #463 already took care of it.

**Pull request closed without merging**